test/unit/test_pos_nfa.js

```diff
@@ -23,18 +23,20 @@ import { toNFA } from '../../lib/pos-parser/nfa';
 
 const TEST_CASES = [
     [
-        '( show me | find me | find | search for ) a [ $value ] $domain', ['restaurant', 'diner'], 'chinese',
+        '( show me | find me | find | search for ) ( a | ε ) [ $value ] $domain', ['restaurant', 'diner'], 'chinese',
```

Comment:
Maybe you should add the `?` operator as syntactic sugar in your regex DSL

Reply:
> I am honestly impressed that you implemented an NFA-based regular expression engine! I would have used backtracking and be done with it...
> The code is very nice, clean and readable, which quite something for a non-trivial algorithm.
> 
> I have a few comments, but overall, let's merge this!

Thanks! The comments are all good - but let's merge this and deal with them later (once tests pass)